drivers/sensor/mcp9600/mcp9600.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022, Maxmillion McLaughlin
+ * Copyright (c) 2020, SER Consulting LLC / Steven Riedl
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_mcp9600
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(MCP9600, CONFIG_SENSOR_LOG_LEVEL);
+
+#define MCP9600_REG_TEMP_HOT 0x00
+
+#define MCP9600_REG_TEMP_DIFF 0x01
+#define MCP9600_REG_TEMP_COLD 0x02
+#define MCP9600_REG_RAW_ADC   0x03
+
+#define MCP9600_REG_STATUS 0x04
+
+#define MCP9600_REG_TC_CONFIG  0x05
+#define MCP9600_REG_DEV_CONFIG 0x06
+#define MCP9600_REG_A1_CONFIG  0x07
+
+#define MCP9600_REG_A2_CONFIG 0x08
+#define MCP9600_REG_A3_CONFIG 0x09
+#define MCP9600_REG_A4_CONFIG 0x0A
+#define MCP9600_A1_HYST	      0x0B
+
+#define MCP9600_A2_HYST	 0x0C
+#define MCP9600_A3_HYST	 0x0D
+#define MCP9600_A4_HYST	 0x0E
+#define MCP9600_A1_LIMIT 0x0F
+
+#define MCP9600_A2_LIMIT	0x10
+#define MCP9600_A3_LIMIT	0x11
+#define MCP9600_A4_LIMIT	0x12
+#define MCP9600_REG_ID_REVISION 0x13
+
+struct mcp9600_data {
+	int32_t temp;
+};
+
+struct mcp9600_config {
+	const struct i2c_dt_spec bus;
+};
+
+static int mcp9600_reg_read(const struct device *dev, uint8_t start, uint8_t *buf, int size)
+{
+	const struct mcp9600_config *cfg = dev->config;
+
+	return i2c_burst_read_dt(&cfg->bus, start, buf, size);
+}
+
+static int mcp9600_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct mcp9600_data *data = dev->data;
+	uint8_t buf[2];
+	int ret;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_AMBIENT_TEMP);
```

Comment:
Why not return ENOTSUP when channel is other than supported to this sensor instead of asserting?

Reply:
I decided to use an assertion instead after reviewing other sensor driver implementations - BH1750 is one of the few I found returning ENOTSUP instead. I thought it would be best to keep it consistent, although I don't really see the benefit.

Reply:
Maybe @MaureenHelm has some thoughts here. I've found many drivers use `-ENOTSUP`, which I think is useful for when one uses the sensor shell commands (i.e it automagically outputs sensor values only for the channels supported)?